src/draft-service.ts

```diff
@@ -362,31 +430,23 @@ export async function getDrafts(): Promise<SubmissionTypes.FormsAppDraft[]> {
  * @param draftId
  * @returns
  */
-export async function getDraftAndData(
-  draftId: string | undefined | null,
-): Promise<{
-  draft: SubmissionTypes.FormsAppDraft
-  draftData: DraftSubmission['submission']
-  lastElementUpdated: FormTypes.FormElement | undefined
-} | null> {
-  if (!draftId) {
-    return null
+async function getDraftAndData(
+  formSubmissionDraftId: string | undefined | null,
+): Promise<DraftSubmission | undefined> {
+  if (!formSubmissionDraftId) {
+    return
   }
 
-  const draft = await getDrafts().then((drafts) =>
-    drafts.find((draft) => draft.draftId === draftId),
-  )
-  if (!draft || !draft.formId || !draft.draftDataId) {
-    return null
+  const localDraftsStorage = await getLocalDrafts()
+  const formSubmissionDraft =
+    localDraftsStorage.syncedFormSubmissionDrafts.find(
+      ({ id }) => id === formSubmissionDraftId,
+    )
+  if (!formSubmissionDraft) {
+    return (await getLocalDraftSubmission(formSubmissionDraftId)) || undefined
   }
 
-  const draftSubmission = await getDraftData(draft.formId, draft.draftDataId)
-
-  return {
-    draftData: draftSubmission.submission,
-    lastElementUpdated: draftSubmission.lastElementUpdated,
-    draft,
-  }
+  return await getDraftSubmission(formSubmissionDraft)
```

Comment:
I know it's current behaviour, but if you've had the app open for ages you may not necessarily have the latest server data.  Unless I'm mistaken.

Reply:
yep yep, will need to be fixed.